### ISSUE TITLE
Can't attach to APS.NET Core application when hostingModel is "InProcess" and .NET CLR version was selected

### ISSUE THREAD

**Issue by ww898** [dotnet/coreclr#26761]
The attach to `w3svc.exe` which contains .NET Core v3.0.100-rc1 inside (ASP.NET Core v3 application where `web.config` has `hostingModel="InProcess"` ) hangs till the timeout when the IIS application pool configured in the following way:
![netf](https://user-images.githubusercontent.com/2767027/65134987-7db1b280-da05-11e9-8955-f55083c7ba31.png).
The server diagnostic thread in `w3svc.exe` is waiting for incoming connection in `ConnectNamedPipe`:
```
  35  Id: 492c.3dd8 Suspend: 1 Teb: 000000b2`c3498000 Unfrozen
 # Child-SP          RetAddr           Call Site
00 000000b2`c467f6c8 00007ff8`91965b99 ntdll!NtFsControlFile+0x14
01 000000b2`c467f6d0 00007ff8`49ac225b KERNELBASE!ConnectNamedPipe+0x79
02 000000b2`c467f740 00007ff8`49ad39a4 coreclr!IpcStream::DiagnosticsIpc::Accept+0x7b [f:\workspace.2\_work\1\s\src\debug\debug-pal\win\diagnosticsipc.cpp @ 64] 
03 000000b2`c467f7a0 00007ff8`953c4034 coreclr!DiagnosticsServerThread+0xa4 [f:\workspace.2\_work\1\s\src\vm\diagnosticserver.cpp @ 51] 
04 000000b2`c467f850 00007ff8`955b3691 KERNEL32!BaseThreadInitThunk+0x14
05 000000b2`c467f880 00000000`00000000 ntdll!RtlUserThreadStart+0x21
```
It looks like the server pipe has wrong access rights in this case. It's the reason of timeout in the client application.

The attach works properly when the .NET CLR version set to "No Managed Code":
![no](https://user-images.githubusercontent.com/2767027/65135247-e1d47680-da05-11e9-9128-a61a55a0a712.png)

The detail repro case is here https://youtrack.jetbrains.com/issue/PROF-903.

**Comment by jeffschwMSFT** [comment]
I do not believe ASP.NET Core 3.0 can target .NET Framework.  See https://docs.microsoft.com/en-us/aspnet/core/?view=aspnetcore-2.2.  +@DamianEdwards 

cc @tommcdon

**Comment by davidfowl** [comment]
@jeffschwMSFT the above works and ends up hosting both CLRs in the same process (coreclr.dll and clr.dll). Are you trying attach to w3svc.exe or w3wp.exe?

**Comment by ww898** [comment]
@davidfowl thanks a lot. You are right, surely `w3wp.exe`.

**Comment by tommcdon** [comment]
Cc @gregg-miskelly

**Comment by davmason** [comment]
@josalem @sywhang have either of you seen this before?

cc @dotnet/dotnet-diag

**Comment by josalem** [comment]
Hmm, assuming that @ww898 is using the IPC server to start profiling, the error message  makes me think that a Named Pipe doesn't exist for that PID.  @ww898, could you try to repro this and then before attaching do `gci \\.\pipe\ | findstr dotnet`.  This will list all the available IPC pipes you can connect to.  They will have a name like `dotnet-diagnostic-<pid>`.  Try comparing the PIDs in that list to the PID being displayed in your UI.  I'm wondering if somehow the Named Pipe is being created with a different PID in the name than that what is being displayed there due to the different pooling models in IIS.

**Comment by davidfowl** [comment]
The named pipe here is an IIS implementation detail, nothing to do with event pipe

**Comment by noahfalk** [comment]
When you launch w3wp.exe with/without the CLR loaded, does that cause it to use a different app-pool, possibly owned by a different user account? In order to connect to the named pipe the process initiating the connection must either be the same user account as the target or have admin rights. There could also be some other security privilege checks involved such as Mandatory Integrity Level checks that would prevent a lower rights process from connecting to a higher rights process.

I think someone will need to setup a repro over here before we've got a more definitive answer, but my current bet is that the root cause will be a windows security check.
